keras_cv/losses/box_3d_regression_loss.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_cv.bounding_box_3d import CENTER_XYZ_DXDYDZ_PHI
+
+
+def l1(y_true, y_pred, sigma=9.0):
+    """Computes element-wise l1 loss."""
+
+    absolute_difference = tf.abs(y_pred - y_true)
+    loss = tf.where(
+        absolute_difference < 1.0 / sigma,
+        0.5 * sigma * absolute_difference**2,
+        absolute_difference - 0.5 / sigma,
+    )
+
+    return tf.reduce_sum(loss, axis=-1)
+
+
+class Box3DRegressionLoss(keras.losses.Loss):
```

Comment:
do you like the name Box3D?  I feel like theres a bit of an inversion with ordering here (3D box is my expected ordering), but I understand python probably prevents the name 3DBox.  

Wonder if there's anything else we can call these components.

No strong opinion, and honestly no better name in mind, just thought I'd mention it in case you have a better anme.

Reply:
Yeah, I wanted to call it `3DBoxRegressionLoss`, but Python doesn't like starting with the `3`

Reply:
(As discussed in 1:1 let’s call it CenterPillarLoss)